frontstage/controllers/conversation_controller.py

```diff
@@ -107,44 +108,83 @@ def send_message(
     if not secure_message_form.validate():
         raise InvalidSecureMessagingForm(secure_message_form.errors)
 
-    message_json = {
-        "msg_from": party_id,
-        "msg_to": msg_to,
-        "subject": subject,
-        "category": category,
-        "body": secure_message_form["body"].data,
-        "thread_id": secure_message_form["thread_id"].data,
-    }
-
-    if survey_id:
-        message_json["survey_id"] = survey_id
-    if business_id:
-        message_json["business_id"] = business_id
-
-    url = f"{current_app.config['SECURE_MESSAGE_URL']}/messages"
+    body = secure_message_form["body"].data
+    thread_id = secure_message_form["thread_id"].data
+    sm_version = current_app.config["SECURE_MESSAGE_VERSION"]
+    session = _get_session()
     headers = _create_send_message_headers()
 
-    with _get_session() as session:
-        response = session.post(url, headers=headers, data=json.dumps(message_json))
+    if sm_version in ("v1", "both"):
+        sm_v1_message_json = {
+            "msg_from": party_id,
+            "msg_to": msg_to,
+            "subject": subject,
+            "category": category,
+            "body": body,
+            "thread_id": thread_id,
+        }
+
+        if survey_id:
+            sm_v1_message_json["survey_id"] = survey_id
+        if business_id:
+            sm_v1_message_json["business_id"] = business_id
+
+        url = f"{current_app.config['SECURE_MESSAGE_URL']}/messages"
+        response = session.post(url, headers=headers, data=json.dumps(sm_v1_message_json))
         try:
             response.raise_for_status()
         except HTTPError:
             logger.error("Message sending failed due to API Error", party_id=party_id, exc_info=True)
             raise ApiError(logger, response)
 
-    msg_id = response.json()["msg_id"]
-
-    logger.info(
-        "Secure message sent successfully",
-        message_id=msg_id,
-        party_id=party_id,
-        category=category,
-        survey_id=survey_id,
-        business_id=business_id,
-        ce_id=ce_id,
-    )
-
-    return msg_id
+        sm_v1_msg_id = response.json()["msg_id"]
+
+        logger.info(
+            "Secure message sent successfully",
+            message_id=sm_v1_msg_id,
+            party_id=party_id,
+            category=category,
+            survey_id=survey_id,
+            business_id=business_id,
+            ce_id=ce_id,
+        )
+
+    if current_app.config["SECURE_MESSAGE_VERSION"] in ("v2", "both"):
+        sm_v2_thread = {
```

Comment:
As discussed this isn't quite right, as this will always create a new thread for every message, which isn't right if replying to ongoing thread, however until the conversation controller is updated fully this is the only feasible solution